src/base-apis.js

```diff
@@ -891,18 +911,25 @@ MatrixBaseApis.prototype.sendStateEvent = function(roomId, eventType, content, s
 /**
  * @param {string} roomId
  * @param {string} eventId
+ * @param {string=} txnId     transaction id. One will be made up if not
+ *    supplied.
  * @param {module:client.callback} callback Optional.
  * @return {module:client.Promise} Resolves: TODO
  * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
-MatrixBaseApis.prototype.redactEvent = function(roomId, eventId, callback) {
-    const path = utils.encodeUri("/rooms/$roomId/redact/$eventId", {
+MatrixBaseApis.prototype.redactEvent = function(
+    roomId, eventId, txnId, callback,
```

Comment:
I suppose as it stands this would be a breaking change, as a parameter is added to a public method. Either we'll need to bump the major version and add a warning to the CHANGELOG, or you shift the callback argument to the left if the txnId is undefined, so it behaves the same as before.

I think I would prefer the latter.

Reply:
Yeah that seems fair to use callback if tnxId is undefined. Do I have to close this pull request and send a new one.

Reply:
Also checking if that's a instance of Funcation may help

Reply:
Just pushing more commits here is fine. I suppose the most correct way would be to check arguments.length, and if it's 3, do `callback = txnId`.

We generally don't check the type of arguments in the SDK fwiw.